docs/source/contributing_roles.rst

```diff
@@ -1,71 +1,227 @@
+    less secure and harder to debug.
+    In most cases there already is a built-in module, or a collection module,
+    providing the same functionality.
+
+Tasks must have a descriptive name, ideally without use of templating.
 
-    (molecule-venv) % cd roles/edpm_<role>
-    (molecule-venv) % molecule test # tests default scenario
-    (molecule-venv) % molecule test --all
-    (molecule-venv) % molecule test --scenario-name <specific scenario>
+Error handling
+++++++++++++++
 
-Testing roles with ansibleee
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Roles must accurately report all failures, with a concise, yet descriptive, message that can point
+administrator in the right direction. Error messages should not include complex data structures,
+like dictionaries or lists, in order to strain on human parser.
 
-edpm-ansible is included in the openstack-ansibleee-runner container image,
-which is key component used by the dataplane-operator that deploys EDPM
-nodes. The dataplane-operator's CRD includes support for specifying additional
-volume mounts for the ansibleee pods, which provides a mechanism for accessing
-a local copy of edpm-ansible. This makes it possible to develop and test local
-changes edpm-ansible without having to build and deploy a new
-openstack-ansibleee-runner container image.
+Privilege escalation
+++++++++++++++++++++
 
-.. toctree::
+`Privilege escalation`_, or `become`, must be used only when neccessary.
+Only tasks that have to be executed under root privileges should receive them.
+Even if there are multiple tasks requiring root privileges, separated by others that can operate
+under normal user, the entire role must not operate under them.
+
+The same applies when using `become` to switch to a different, non-root, user.
+
+Special care must be taken when using `become` with modules executing scripts,
+like the `ansible.builtin.cmd`. And when invoking other roles.
+
+Role test development
+~~~~~~~~~~~~~~~~~~~~~
+
+Roles must have molecule tests with at least one scenario representing the most common
+expected usage. Tests must be included in the CI layout and executed after every
+change that may conceivably affect role function.
+
+Molecule tests should be generated with tooling provided in the repository, or derived
+from a test suite of an existing role.
+
+Tests should utilize either podman, or delegated driver, with podman given precedence.
+Delegated driver, or other drivers, should be used only when it isn't possible to properly
+test the role inside of a container.
+
+Roles with multiple execution paths should have multiple scenarios. All molecule scenarios,
+apart from the default, must have a descriptive name.
+
+
+Documenting a new role
+~~~~~~~~~~~~~~~~~~~~~~
+
+Roles must accurately explain their effects in their documentation, with in-line
+comments serving only as a supplement. In complex cases the documentation should link
+to other material, but only as a source of additional information. Beware the link rot.
```

Comment:
'Beware the link rot.' leaves a lot to interpretation.

Perhaps it's worth expanding a little and saying something like 'Care should be taken to use permanent links, especially when referencing something versioned or a git repository code reference.'

Reply:
I actually think it's better to leave this open to interpretation. It doesn't burden people too much while still giving guidance. Of course, if you are adamant about it, I can change it to something else.